gridftp/server/src/modules/file/globus_gridftp_server_file.c

```diff
@@ -1090,7 +1095,7 @@ globus_l_gfs_file_stat(
         if(!stat_array)
         {
             result = GlobusGFSErrorMemory("stat_array");
-            goto error_alloc2;
+            goto error_open;
```

Comment:
And why can the goto labels be changed in that way (also further below)?

Reply:
The code was:

    error_alloc2:
        closedir(dir);
    
    error_open:

But dir is only used in the #ifdef WIN32 block. In the #else part goto error_alloc2 does not make sense. It should skip the closedir and goto error_open instead.

Reply:
That makes sense. Thanks for the explanation.